Jellyfin.Api/Models/NotificationDtos/NotificationDto.cs

```diff
@@ -0,0 +1,53 @@
+#nullable enable
+
+using System;
+using MediaBrowser.Model.Notifications;
+
+namespace Jellyfin.Api.Models.NotificationDtos
+{
+    /// <summary>
+    /// The notification DTO.
+    /// </summary>
+    public class NotificationDto
+    {
+        /// <summary>
+        /// Gets or sets the notification ID. Defaults to an empty string.
+        /// </summary>
+        public string Id { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the notification's user ID. Defaults to an empty string.
+        /// </summary>
+        public string UserId { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the notification date.
+        /// </summary>
+        public DateTime Date { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the notification has been read. Defaults to false.
+        /// </summary>
+        public bool IsRead { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets the notification's name. Defaults to an empty string.
+        /// </summary>
+        public string Name { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the notification's description. Defaults to an empty string.
+        /// </summary>
+        public string Description { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the notification's URL. Defaults to null.
+        /// </summary>
+        public string? Url { get; set; }
```

Comment:
This is a non-nullable string in the old service. Why change it?

Reply:
With the new API, the way to mark a parameter as optional is to make it nullable in the controller action which handles it:
```csharp
[HttpPost("Admin")]
        public void CreateAdminNotification(
            [FromForm] string name,
            [FromForm] string description,
            [FromForm] string? url,
            [FromForm] NotificationLevel? level)
```
and in order to assign that nullable string to the DTO property, it has to be nullable on the DTO.